src/__experimental__/components/grouped-character/grouped-character.component.js

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import Textbox from '../textbox';
+import { generateGroups, toSum, toIndexSteps } from './grouped-character.utils';
+import Events from '../../../utils/helpers/events/events';
+
+const GroupedCharacter = (props) => {
+  const [pressedKey, updatePressedKey] = useState(null);
+
+  const { separator, groups, value } = props;
+  const stepIndices = groups.reduce(toIndexSteps, []);
+
+  const handleChange = (e) => {
+    const i = e.target;
+    const { selectionEnd } = e.target;
+    let modifier = 0;
+
+    if (stepIndices.includes(selectionEnd - 1)) modifier = 1;
```

Comment:
If you switched these you could use an `if/ else if` and only adjust the `modifier` value once?

```
 if (stepIndices.includes(selectionEnd - 1) && Events.isBackspaceKey({ which: pressedKey })) modifier = -1;
 else if (stepIndices.includes(selectionEnd - 1)) modifier = 1;
```

Reply:
don't you think that would be a bit difficult to read

Reply:
I would extract conditionals to consts and then:
```
if(hasSelectionEnd) {
 isBackspacePressed ? selectionEnd-- : selectionEnd++;
}
```